### PR TITLE
CI migration to gh actions + Black formatting

### LINKED ISSUE #115: Code formatting? Perhaps black to match xgcm
I'm suggesting here something similar to https://github.com/xgcm/xgcm/pull/131

### PR DESCRIPTION
**Pull request by jbusecke**
This PR moves the CI to github actions and implements linting with black as part of the CI.
- [x] Closes #115

### CODE REVIEW

**Comment by jbusecke** [comment]
@xgcm/xrft-dev Can one of the admins enable RTD builds for PR's following these instructions?

https://docs.readthedocs.io/en/stable/guides/autobuild-docs-for-pull-requests.html

That would help tremendously with making changes to the docs.

**Comment by jbusecke** [comment]
I think thats it. All the tests pass with the new CI and black formatting is enforced via `pre-commit`. I also added a `contributor-guide` to the docs. Let me know if you think this needs further changes.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache conda
+        uses: actions/cache@v1
+        env:
+          # Increase this value to reset cache if ci/environment.yml has not changed
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/environment-py${{ matrix.python-version }}.yml') }}
```
Comment (rabernat) [inline .github/workflows/ci.yaml:30]: What is this warning?

**Review by rabernat — COMMENTED** [review]
This is fantastic Julius! Thanks so much for all this work.

Just one question...

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jbusecke) [reply .github/workflows/ci.yaml:30]: I am not actually sure. I copied this from another package, pretty sure either @dcherian or @andersy005 know more about it.

Comment (rabernat) [reply .github/workflows/ci.yaml:30]: This is what I see:

image

I think it is the [Chrome MathJax Plugin](https://chrome.google.com/webstore/detail/mathjax-plugin-for-github/ioemnmodlmafdkllaclgeombjnmnbima?hl=en) screwing things up for me!

So this is fine. The workflow works.

**Pull request merged**